kubebuilder/Tiltfile

```diff
@@ -0,0 +1,64 @@
+load('ext://restart_process', 'docker_build_with_restart')
+
+def kubebuilder(DOMAIN, GROUP, VERSION, KIND, IMG='controller:latest'):
+
+    DOCKERFILE = '''FROM golang:alpine
+    WORKDIR /
+    COPY ./bin/manager /
+    CMD ["/manager"]
+    '''
+
+    def yaml():
+        return local('cd config/manager; kustomize edit set image controller=' + IMG + '; cd ../..; kustomize build config/default')
+    
+    def manifests():
+        return 'controller-gen crd:trivialVersions=true rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases;'
+    
+    def generate():
+        return 'controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./...";'
+    
+    def vetfmt():
+        return 'go vet ./...; go fmt ./...'
+    
+    def binary():
+        return 'CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -o bin/manager main.go'
+
+    installed = local("which kubebuilder")
+    print("kubebuilder is present:", installed)
+
+    DIRNAME = os.path.basename(os. getcwd())
+    initting = False
+    if os.path.exists('go.mod') == False:
+        initting = True
+        local("go mod init %s" % DIRNAME)
+    
+    if os.path.exists('PROJECT') == False:
+        initting = True
+        local("kubebuilder init --domain %s" % DOMAIN)
+    
+    if os.path.exists('api') == False:
+        initting = True
+        local("kubebuilder create api --resource --controller --group %s --version %s --kind %s" % (GROUP, VERSION, KIND))
+    
+    local(manifests() + generate())
+    
+    local_resource('CRD', manifests() + 'kustomize build config/crd | kubectl apply -f -', deps=["api"])
```

Comment:
*sigh* i wish we had a better answer for this

Reply:
WDYM?

Reply:
tilt can't track resources that are applied with `kubectl apply`. But the primitives for dynamically generating YAML also aren't great. There's some discussion on it here:
https://github.com/tilt-dev/tilt/issues/2664
https://github.com/tilt-dev/tilt/issues/3046

DM and I had spent a bunch of time thinking about what a better solution to this might be...it's a problem that a lot of build systems wrestle with...